pkg/provisioner/api/provisioner.go

```diff
@@ -26,7 +26,15 @@ import (
 // methods used to create and delete new buckets, and to grant or revoke access
 // to buckets within the object store.
 type Provisioner interface {
-	// Provision should be implemented to handle bucket creation
+	// Provision should be implemented to handle creation of object storage buckets.
+	// Provision should NOT create the ObjectBucket resource. ObjectBucket resource creation is done
+	// by this library's controller.
+	// The Provision implementation must return an ObjectBucket struct with at least the Connection
+	// spec filled in. All other ObjectBucket details will be filled in by this library's controller
+	// before the ObjectBucket resource is created.
+	// The Provision implementation may opt to specify the ObjectBucket spec's ReclaimPolicy in
+	// cases where the provisioner wishes to set a different value from the one specified in the
+	// ObjectBucketClaim's StorageClass.
```

Comment:
@jeffvance I have updated the API documentation for `Provision` to clarify more details about how it should be implemented. Let me know if I should also (or alternately) update info in other places.